File: app/src/main/java/com/pcsalt/example/productflavors/MainActivity.java
package com.pcsalt.example.productflavors;

import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.pcsalt.example.productflavors.utils.MyLog;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView textView = (TextView) findViewById(R.id.text_view);

        textView.setText(getString(R.string.app_name) + " - " + BuildConfig.VERSION_NAME);

        MyLog.e(TAG, "some error log");
    }
}

File: app/src/main/kotlin/com/pcsalt/example/productflavors/MainActivity.kt
package com.pcsalt.example.productflavors

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.pcsalt.example.productflavors.databinding.ActivityMainBinding
import com.pcsalt.example.productflavors.utils.MyLog

class MainActivity : AppCompatActivity() {

  private lateinit var binding: ActivityMainBinding

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    binding = ActivityMainBinding.inflate(layoutInflater)
    setContentView(binding.root)

    binding.textView.text = "${getString(R.string.app_name)} - ${BuildConfig.VERSION_NAME}"

    MyLog.e(TAG, "some error log")
  }

  companion object {
    private const val TAG = "MainActivity"
  }
}
